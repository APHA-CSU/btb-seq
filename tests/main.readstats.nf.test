nextflow_process {

    name "Test Process readStats"
    script "main.nf"
    process "READSTATS"

    test("Should generate stats on data processing") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                input[0] = ["phix-m", file("${projectDir}/tests/data/phix_test/phix-m_S800_R1_001.fastq.gz"), 
                            file("${projectDir}/tests/data/phix_test/phix-m_S800_R2_001.fastq.gz"), 
                            file("${projectDir}/tests/data/phix_test/dedup_1.fastq"), 
                            file("${projectDir}/tests/data/phix_test/dedup_2.fastq"), 
                            file("${projectDir}/tests/data/phix_test/trimmed_1.fastq"),
                            file("${projectDir}/tests/data/phix_test/trimmed_2.fastq"),
                            file("${projectDir}/tests/data/phix_test/phix-800.bam"),]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}