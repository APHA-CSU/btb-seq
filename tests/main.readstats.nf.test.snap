{
    "Should generate stats on data processing": {
        "content": [
            {
                "0": [
                    [
                        "phix-m",
                        "phix-m_stats.csv:md5,d6cd5066345828d9dd1ecf1afc498ba1"
                    ]
                ],
                "1": [
                    [
                        "phix-m",
                        "phix-m_outcome.txt:md5,fbb6e44cb0ecb2c8284885dae762318d"
                    ]
                ],
                "outcome": [
                    [
                        "phix-m",
                        "phix-m_outcome.txt:md5,fbb6e44cb0ecb2c8284885dae762318d"
                    ]
                ],
                "stats": [
                    [
                        "phix-m",
                        "phix-m_stats.csv:md5,d6cd5066345828d9dd1ecf1afc498ba1"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-03T16:21:56.621229756"
    }
}